### ISSUE TITLE
making cms_pico sites available in "root folders" (own domains )

### ISSUE THREAD

**Issue by testbird** [nextcloud/cms_pico#57]
There doesn't seem to be an option (nor documented manual config currently) that supports to make a cms_pico site in nextcloud available as the "root folder" of a domain.

Either
1) In the root of the main nextcloud domain (and the Nextcloud interface only being available in a subfolder, like for example /login ).
2) Under separate domains used to access specific cms_pico sites.

Until there is a configuration option, it would be good to document a manual solution at:
https://github.com/nextcloud/cms_pico/wiki/Admin-Settings#apache-configuration

**Comment by weipah** [comment]
Just some thoughts...
I think to use a PicoCMS website in the root you would have to move your NC installation to a subfolder first, like example.org/nextcloud/. Then you can work with proxy/rewrite settings in apache to map PicoCMS to the root.

For different domains it should be no problem, because your NC installation should not interfere.

**Comment by PhrozenByte** [comment]
Pico CMS for Nextcloud v1.0.0 is arriving and I'm currently working through some old issues that can be resolved before releasing the app's first stable release. See [help.nextcloud.com](https://help.nextcloud.com/t/beta-needs-testing-pico-cms-for-nextcloud-v1-0-0/58718) for more info.

This should be fixed in Pico CMS for Nextcloud v1.0.0. You can check it out now to test whether your issue persists or not. If it does, please open a new Issue here on GitHub.

**Comment by brasofilo** [comment]
I adapted @julianwachholz solution at [issue#83](https://github.com/nextcloud/cms_pico/issues/83#issuecomment-589543921) (for subdomains) to use a top level domain. 

Here, `newdomain.com` is the own domain for Pico folder/site `website-newdomain` hosted on domain `nextcloud.domain.com`. This considers that Nextcloud is already configured with short URLs.

```nginx
server {
  listen 443 ssl;
  listen [::]:443 ssl;

  server_name newdomain.com;

  include /config/nginx/ssl.conf;

  # required for theme files (css etc)
  location /apps/cms_pico/ {
    proxy_pass https://nextcloud.domain.com;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }

  location / {
    proxy_pass https://nextcloud.domain.com/apps/cms_pico/pico_proxy/website-newdomain/;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

    # don't pass unnecessary nextcloud cookies
    proxy_hide_header "set-cookie";
    
    # rewrite links to keep to the domain root
    sub_filter "/sites/website-newdomain/" "/";
    sub_filter_once off;
  }
}
```
